File: 1608. Special Array With X Elements Greater Than or Equal X.rb
#* 1608. Special Array With X Elements Greater Than or Equal X
nums = [3,5]
# @param {Integer[]} nums
# @return {Integer}
def special_array(nums)
  i = nums.size
  while i > 0
    return i if i == nums.count{|num| num >= i }
    i-=1
  end
  -1
end
p special_array(nums)

File: 1608. Special Array With X Elements Greater Than or Equal X.py
# PYTHON
#* 1608. Special Array With X Elements Greater Than or Equal X
nums = [3,5]
#*
def specialArray(nums):
  i = len(nums)
  while i > 0:
      if i == sum(1 for num in nums if num >= i):
          return i
      i -= 1
  return -1
  
print(specialArray(nums))


#* WORKING . . .
#* NO GOOD
def specialArray(nums):
  i = len(nums)
  while i > 0:
    return sum(1 for num in nums if num >= i)


#* MISC
# Ruby:
# nums = [1, 2, 3, 4, 5]
# i = 3
# count = nums.count { |num| num >= i }  # This will be 3 (elements: 3, 4, 5)
# puts count

# Python:
nums = [1, 2, 3, 4, 5]
i = 3
count = sum(1 for num in nums if num >= i)  # This will be 3 (elements: 3, 4, 5)
print(count)
